File: springboot_mp/src/main/java/com/xeon/springboot_mp/config/MyBatisPlusConfig.java
package com.xeon.springboot_mp.config;


import com.baomidou.mybatisplus.annotation.DbType;
import com.baomidou.mybatisplus.extension.plugins.MybatisPlusInterceptor;
import com.baomidou.mybatisplus.extension.plugins.inner.PaginationInnerInterceptor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;


@Configuration
public class MyBatisPlusConfig {

    @Bean
    public MybatisPlusInterceptor mybatisPlusInterceptor() {

        System.out.println("MybatisPlusInterceptor 执行");
        final MybatisPlusInterceptor mybatisPlusInterceptor = new MybatisPlusInterceptor();
        /*分页*/
        mybatisPlusInterceptor.addInnerInterceptor(new PaginationInnerInterceptor(DbType.MYSQL));


        return mybatisPlusInterceptor;
    }


}

File: kotlin_springboot_maven/src/main/kotlin/com/yzq/kotlin_springboot_maven/config/MyBatisPlusConfig.kt
package com.yzq.kotlin_springboot_maven.config

import com.baomidou.mybatisplus.annotation.DbType
import com.baomidou.mybatisplus.extension.plugins.MybatisPlusInterceptor
import com.baomidou.mybatisplus.extension.plugins.inner.PaginationInnerInterceptor
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration


/**
 * @description: Mybatis plus的一些配置
 * @author : yuzhiqiang (zhiqiang.yu.xeon@gmail.com)
 */
@Configuration
class MyBatisPlusConfig {

    @Bean
    fun mybatisPlusInterceptor(): MybatisPlusInterceptor {
        val mybatisPlusInterceptor = MybatisPlusInterceptor()
        /*分页*/
        mybatisPlusInterceptor.addInnerInterceptor(PaginationInnerInterceptor(DbType.MYSQL))
        return mybatisPlusInterceptor
    }

}